sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -124,6 +124,32 @@ RT::PiProgram ProgramManager::getClProgramFromClKernel(RT::PiKernel Kernel) {
   return Program;
 }
 
+string_class ProgramManager::getProgramBuildLog(const RT::PiProgram &Program) {
+  size_t Size = 0;
+  PI_CALL(RT::piProgramGetInfo(Program, CL_PROGRAM_DEVICES, 0, nullptr, &Size));
+  vector_class<RT::PiDevice> PIDevices(Size / sizeof(RT::PiDevice));
+  PI_CALL(RT::piProgramGetInfo(Program, CL_PROGRAM_DEVICES, Size,
+                               PIDevices.data(), nullptr));
+  string_class Log = "The program was built for " +
```

Comment:
Technically if you here then program is not built for at least one of the devices.